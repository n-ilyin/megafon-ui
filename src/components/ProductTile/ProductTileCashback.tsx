import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileCashback.less';
import { ICashback } from './types';
import { productTileCashbackProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-cashback');
class ProductTileCashback extends React.Component<Partial<ICashback>> {
    static propTypes = productTileCashbackProps;

    render() {
        const { title, value, unit } = this.props;

        if (!title && !value && !unit) {
            return null;
        }

        return (
            <div className={cn('')}>
                <span className={cn('text')}>
                    {title} {value} {unit}
                </span>
            </div>
        );
    }
}

export default ProductTileCashback;
