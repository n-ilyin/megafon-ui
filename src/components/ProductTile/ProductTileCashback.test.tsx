import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTileCashback from './ProductTileCashback';

const props = {
    title: 'Кэшбэк',
    value: 23,
    unit: '%',
};

describe('<ProductTileCashback />', () => {
    it('renders ProductTileCashback', () => {
        const wrapper = shallow(<ProductTileCashback {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should not renders without props', () => {
        const wrapper = shallow(<ProductTileCashback />);

        expect(wrapper.type()).toEqual(null);
    });
});
