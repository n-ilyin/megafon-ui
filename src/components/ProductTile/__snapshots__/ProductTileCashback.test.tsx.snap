// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTileCashback /> renders ProductTileCashback 1`] = `
<div
  className="mfui-product-tile-cashback"
>
  <span
    className="mfui-product-tile-cashback__text"
  >
    Кэшбэк
     
    23
     
    %
  </span>
</div>
`;
